### DISCUSSION TITLE
Can I get those Quick data points in HNSW method in Milvus  and alternate it?

### DISCUSSION THREAD

**Discussion by Kevin-shihello-world** [milvus-io/milvus#24840] [Q&A and General discussion]
-  As what showed in the title, we know that hnsw method do its job by first search those quick data points. So it those points may not be always suitable for all those tasks . So I wonder if we can get those points and alternate them? 

I would be appreciate if anyone offer me an answer  Thank you for your time.( A friendly laugh)

**Comment by yhmo** [comment]
The underlying engine of Milvus is Knowhere: https://github.com/milvus-io/knowhere
Milvus calls knowhere interfaces to search and build index, but the index inner data is invisible to Milvus.
Knowhere can be installed as a python lib and tested by python script: https://github.com/milvus-io/knowhere/blob/main/tests/python/test_index_random.py
You can browse the interfaces of knowhere: https://github.com/milvus-io/knowhere/blob/main/python/knowhere/__init__.py, I am not sure is there an interface that can meet your requirement.

Another related project is the Fader: https://github.com/zilliztech/feder, this is a front-end project to visualize index(including the HNSW). Mainter is inactive now, you might need to go deep into the project by yourself.
